pyEpiabm/pyEpiabm/tests/test_unit/test_sweep/test_spatial_sweep.py

```diff
@@ -14,69 +15,134 @@ class TestSpatialSweep(TestMockedLogs):
-    def test_find_infectees(self, mock_dist, mock_nan, mock_logger):
+    def test_find_infectees_successful(self, mock_dist, mock_nan, mock_logger):
         Parameters.instance().infection_radius = 1000
         test_pop = self.pop
         test_sweep = SpatialSweep()
+        mock_dist.return_value = 2.2
         test_sweep.bind_population(test_pop)
-        test_pop.add_cells(1)
-        cell_susc = test_pop.cells[1]
-        cell_susc.add_microcells(1)
-        microcell_susc = cell_susc.microcells[0]
-        microcell_susc.add_people(1)
-        infectee = microcell_susc.persons[0]
-
-        # Check when all (one) nan in distance, won't call nan_to_num
-        mock_dist.return_value = 0
-        test_list = test_sweep.find_infectees(self.cell_inf, [cell_susc], 1)
-        self.assertFalse(mock_nan.called)
-        self.assertEqual(test_list, [infectee])
-
-        mock_dist.return_value = 2
-        test_list = test_sweep.find_infectees(self.cell_inf, [cell_susc], 1)
+        test_list = test_sweep.\
+            find_infectees(self.cell_inf, [self.cell_susc], 1)
         self.assertFalse(mock_nan.called)
-        self.assertEqual(test_list, [infectee])
+        self.assertEqual(test_list, [self.infectee])
 
-        mock_nan.return_value = [2, 0.0000001]
-        mock_dist.side_effect = [0, 2]
-        test_pop.add_cells(1)
-        third_cell = test_pop.cells[2]
-        third_cell.add_microcells(1)
-        third_cell.microcells[0].add_people(1)
+    @mock.patch("logging.exception")
+    @mock.patch("numpy.nan_to_num")
+    @mock.patch("pyEpiabm.utility.DistanceFunctions.dist_euclid")
+    def test_find_infectees_fails(self, mock_dist, mock_nan, mock_logger):
+        Parameters.instance().infection_radius = 0.0001
+        # Assert a basic population
+        test_pop = self.pop
+        test_sweep = SpatialSweep()
+        mock_dist.return_value = 2.2
+        test_sweep.bind_population(test_pop)
 
-        test_list = test_sweep.find_infectees(self.cell_inf, [cell_susc,
-                                                              third_cell], 1)
-        mock_nan.assert_called_once_with([np.nan, 0.5], nan=0.5)
-        self.assertEqual(test_list, [infectee])
+        self.assertEqual(self.cell_inf.nearest_neighbours, {})
 
         # test the assert that the distance weights has correct length
-        mock_dist.side_effect = [0, 2]
-        mock_nan.return_value = [1]
-        self.assertRaises(AssertionError, test_sweep.find_infectees,
-                          self.cell_inf, [cell_susc, third_cell], 1)
+        # mock_dist.side_effect = [0, 2]
+        # mock_nan.return_value = [1]
+        # self.assertRaises(AssertionError, test_sweep.find_infectees(
+        #                   self.cell_inf, [self.cell_susc], 1))
 
         # test value error is raised if all cells too far away
         Parameters.instance().infection_radius = 0.000001
         mock_dist.side_effect = [0, 2]
         mock_nan.return_value = [1, 1]
```

Comment:
Sorry, I'm not sure what's meaning for these values. L124 sets infection_radius to 0.0001 with mock_dist.return_value = 2.2, will 0.000001 differs the result with 0.0001 for infection_radius? What's side_effect and return_value here?